Tests/tests/test_SystemServices.cpp

```diff
@@ -1306,3 +1307,445 @@ TEST_F(SystemServicesEventIarmTest, onRebootRequest)
+    EXPECT_EQ(response, string("{\"hardwareID\":\"5678\",\"success\":true}"));
+
+}
+
+/**
+ * @brief : getDeviceInfo returns Cached data When QueryParam is ModelName
+ *          Check (i) If input query param is  device's ModelName
+ *          and (ii) if cached data of  Model Name is available
+ *          then , getDeviceInfo shall successfully retrieves the cached information and returns it in the response
+ * @param[in]   :  "params": {"params": "modelName"}
+ * @return      :  {"modelName":"IP061-ec","success":true}
+ */
+TEST_F(SystemServicesTest, getDeviceInfoSuccess_getCachedModelName)
+{
+   //Below IARM_Bus_Call function is called for saving the retrieved data
+   //in member variables [cached value] & setting the corresponding flags to true
+   ON_CALL(iarmBusImplMock, IARM_Bus_Call)
+        .WillByDefault(
+            [](const char* ownerName, const char* methodName, void* arg, size_t argLen) {
+                EXPECT_EQ(string(ownerName), string(_T(IARM_BUS_MFRLIB_NAME)));
+                EXPECT_EQ(string(methodName), string(_T(IARM_BUS_MFRLIB_API_GetSerializedData)));
+                auto* param = static_cast<IARM_Bus_MFRLib_GetSerializedData_Param_t*>(arg);
+                const char* str = "IP061-ec";
+                param->bufLen = strlen(str);
+                strncpy(param->buffer, str, sizeof(param->buffer));
+                param->type =  mfrSERIALIZED_TYPE_SKYMODELNAME;
+                return IARM_RESULT_SUCCESS;
+            });
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getDeviceInfo"), _T("{\"params\":modelName}"), response));
+    EXPECT_EQ(response, string("{\"modelName\":\"IP061-ec\",\"success\":true}"));
+
+    //To confirm that the retrieved data is cached Data;
+    //sets an expectation that the IARM_Bus_Call function should not be called during this sequence
+    EXPECT_CALL(iarmBusImplMock, IARM_Bus_Call)
+        .Times(0)
+        .WillRepeatedly(
+            [](const char* ownerName, const char* methodName, void* arg, size_t argLen) {
+                EXPECT_EQ(string(ownerName), string(_T(IARM_BUS_MFRLIB_NAME)));
+                EXPECT_EQ(string(methodName), string(_T(IARM_BUS_MFRLIB_API_GetSerializedData)));
+                return IARM_RESULT_SUCCESS;
+            });
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getDeviceInfo"), _T("{\"params\":modelName}"), response));
+    EXPECT_EQ(response, string("{\"modelName\":\"IP061-ec\",\"success\":true}"));
+}
+
+/**
+ * @brief : getDeviceInfo  When QueryParam passed without label "param"
+ *          Check if QueryParams  contains no label as "params"
+ *          then getDeviceInfo shall successfully retrieve the device info and  returns it in the response
+ * @param[in]   :   "params" :{}
+ * @return      : {"make":"SKY","success":true}
+ */
+TEST_F(SystemServicesTest, getDeviceInfoSuccess_onQueryParameterHasNoLabelParam)
```

Comment:
Shouldn't this return all the parameters if no param is passed. I see we are checking only for make?